backend/src/main/java/org/sefglobal/core/academix/service/CategoryService.java

```diff
@@ -27,4 +31,19 @@ public CategoryService(CategoryRepository categoryRepository) {
     public List<CustomCategory> getAllCategories(){
         return categoryRepository.findAllBy();
     }
+
+    /**
+     * Retrieves the category with the requested id
+     * @param categoryId which is the id of the requested category
+     * @return Category object
+     * @throws ResourceNotFoundException if a category with the requested id doesn't exist
+     */
+    public Category getCategoryById(Long categoryId) throws ResourceNotFoundException {
+        if(!categoryRepository.existsById(categoryId)){
+            String msg = "Error, Category by id:"+categoryId+" doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        }
+        return categoryRepository.getOne(categoryId);
```

Comment:
You're querying a category twice here by `categoryRepository.existsById(categoryId)` and `categoryRepository.getOne(categoryId)` which is unnecessary.
Instead, you can use `findById` method and return an `Optional` and then validate that.

Reply:
Created an `Optional<CustomCategory>` variable and validated it. But the `findById` method doesn't return a `CustomCategory`. So i created `findBy_Id` method in `CategoryRepository`